src/main/java/teammates/storage/sqlentity/FeedbackQuestion.java

```diff
@@ -0,0 +1,275 @@
+    }
+
+    public void setQuestionType(FeedbackQuestionType questionType) {
+        this.questionType = questionType;
+    }
+
+    public String getQuestionText() {
+        return questionText;
+    }
+
+    public void setQuestionText(String questionText) {
+        this.questionText = questionText;
+    }
+
+    public FeedbackParticipantType getGiverType() {
+        return giverType;
+    }
+
+    public void setGiverType(FeedbackParticipantType giverType) {
+        this.giverType = giverType;
+    }
+
+    public FeedbackParticipantType getRecipientType() {
+        return recipientType;
+    }
+
+    public void setRecipientType(FeedbackParticipantType recipientType) {
+        this.recipientType = recipientType;
+    }
+
+    public Integer getNumOfEntitiesToGiveFeedbackTo() {
+        return numOfEntitiesToGiveFeedbackTo;
+    }
+
+    public void setNumOfEntitiesToGiveFeedbackTo(Integer numOfEntitiesToGiveFeedbackTo) {
+        this.numOfEntitiesToGiveFeedbackTo = numOfEntitiesToGiveFeedbackTo;
+    }
+
+    public List<FeedbackParticipantType> getShowResponsesTo() {
+        return showResponsesTo;
+    }
+
+    public void setShowResponsesTo(List<FeedbackParticipantType> showResponsesTo) {
+        this.showResponsesTo = showResponsesTo;
+    }
+
+    public List<FeedbackParticipantType> getShowGiverNameTo() {
+        return showGiverNameTo;
+    }
+
+    public void setShowGiverNameTo(List<FeedbackParticipantType> showGiverNameTo) {
+        this.showGiverNameTo = showGiverNameTo;
+    }
+
+    public List<FeedbackParticipantType> getShowRecipientNameTo() {
+        return showRecipientNameTo;
+    }
+
+    public void setShowRecipientNameTo(List<FeedbackParticipantType> showRecipientNameTo) {
+        this.showRecipientNameTo = showRecipientNameTo;
+    }
+
+    public Instant getCreatedAt() {
+        return createdAt;
+    }
+
+    public void setCreatedAt(Instant createdAt) {
+        this.createdAt = createdAt;
+    }
+
+    public Instant getUpdatedAt() {
+        return updatedAt;
+    }
+
+    public void setUpdatedAt(Instant updatedAt) {
+        this.updatedAt = updatedAt;
+    }
+
+    @Override
+    public String toString() {
+        return "Question [id=" + id + ", questionNumber=" + questionNumber + ", description=" + description
+                + ", questionType=" + questionType
+                + ", questionText=" + questionText + ", giverType=" + giverType + ", recipientType=" + recipientType
+                + ", numOfEntitiesToGiveFeedbackTo=" + numOfEntitiesToGiveFeedbackTo + ", showResponsesTo="
+                + showResponsesTo + ", showGiverNameTo=" + showGiverNameTo + ", showRecipientNameTo="
+                + showRecipientNameTo + ", isClosingEmailEnabled=" + ", createdAt=" + createdAt + ", updatedAt="
+                + updatedAt + "]";
+    }
+
+    @Override
+    public int hashCode() {
+        // FeedbackQuestion ID uniquely identifies a notification.
+        return this.id.hashCode();
```

Comment:
This doesn't fulfil the contract of hashCode since id can be null until it is generated by the db. This means the same object can have different hash values depending on whether the id has been generated.

Since Feedback Question doesn't have a natural key that we can use, we have two options.

1) Set the hashcode to a fixed constant value
2) Generate the id before persisting to the db

I think the second option is the way to go here, can follow the implementation of ID (UUID) for notification.

Reply:
Okay, got it! i've made the change to use UUID